### PR TITLE
Use official Docker installer and upgrade for :latest installs

### PR DESCRIPTION
**Pull request by cazzerson**
A first attempt at updating the Docker installer as discussed in #6075.

### CODE REVIEW

File: plugins/provisioners/docker/cap/debian/docker_install.rb [hunk 1]
```diff
@@ -4,24 +4,14 @@ module Cap
       module Debian
         module DockerInstall
           def self.docker_install(machine, version)
-            package = 'lxc-docker'
+            package = 'docker-engine'
```
Comment (sethvargo) [inline plugins/provisioners/docker/cap/debian/docker_install.rb]: It doesn't look like this is used anymore?

**Comment by sethvargo** [comment]
Hi @cazzerson 

This is looking really good. I have a few high-level comments:

1. What do you think about extracting the installation into a library like we do in the [Chef installer](https://github.com/mitchellh/vagrant/blob/master/plugins/provisioners/chef/installer.rb)?

1. I think we need to run `apt-get update -qq` before installing curl

1. I think we want to add `-qq` to all the apt calls

I'm not super familiar with the new Docker install script. It's on my list of things to review, but :hourglass: :smile:. Are there options that users would like to pass in? If so, we should provide a way to pass those args in. See the Chef installer for an example.

**Comment by cazzerson** [comment]
@sethvargo Thanks, I'll push those fixes. Two things:

1. Of course, the more I look at this, the more I notice. This may eliminate the ability to freeze the Docker version (defaulting to :latest). This was already the case for RHEL, though. Also, when you have time, search for 'Warning: the "docker" command appears to already exist on this system.' on https://get.docker.com. The script waits for 20 seconds for the user to bail on an upgrade with Ctrl-C. Is there anything else like this in other provisioners?

2. Command line options don't seem to be an issue.

3. I don't have much experience with Chef, but I can take a look when I have :hourglass: :smile:.

**Comment by cazzerson** [comment]
@sethvargo Sorry, not with Chef, **like** Chef :-P I'll take a look.

**Comment by cazzerson** [comment]
@sethvargo Could you tell more about about what ideas you'd like to bring from the Chef installer into the [Docker Installer](https://github.com/mitchellh/vagrant/blob/master/plugins/provisioners/docker/installer.rb)?

**Comment by sethvargo** [comment]
Hey @cazzerson yea sure, but it might be a bit until I have a chance to context switch into that :smile:

**Comment by mitchellh** [comment]
This looks good. My only concern is that it appears installing specific versions has been removed. If we can support that again then I'll merge!

**Comment by cazzerson** [comment]
The first issue seems to be that get.docker.com is no longer recommended--is this correct, @jfrazelle? Additionally, as of 1.9, although the docs recommend uninstalling lxc-docker, the lxc-docker packages now seem to be installing 1.9 on when provisioning a fresh machine. We could switch from lxc-docker to docker-engine (with some check for distro repo selection), but I would need to do more testing for existing installations--I ran into some weird issues when trying to replace lxc-docker by hand.

**Pull request closed without merging**